clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy

```diff
@@ -438,33 +443,55 @@ class DockerRegistryClient {
       catalog.repositories = catalog.repositories.findAll { it ==~ repositoriesRegex }
     }
     if (nextPath) {
-      def nextCatalog = getCatalog(nextPath)
+      def nextPathNew
+      (nextPathNew, queryParams) = parseForQueryParams(nextPath)
+      def nextCatalog = getCatalog(nextPathNew, queryParams)
       catalog.repositories.addAll(nextCatalog.repositories)
     }
 
     return catalog
   }
 
-  public DockerRegistryTags getTags(String repository, String path = null) {
+  public DockerRegistryTags getTags(String repository, String path = null, Map<String, String> queryParams = [:]) {
     def response = request({
-      path ? Retrofit2SyncCall.executeCall(registryService.get(path, tokenService.basicAuthHeader, userAgent)) :
-        Retrofit2SyncCall.executeCall(registryService.getTags(repository, tokenService.basicAuthHeader, userAgent))
+      path ? Retrofit2SyncCall.executeCall(registryService.get(path, tokenService.basicAuthHeader, userAgent, queryParams)) :
+        Retrofit2SyncCall.executeCall(registryService.getTags(repository, tokenService.basicAuthHeader, userAgent, queryParams))
     }, { token ->
-      path ? Retrofit2SyncCall.executeCall(registryService.get(path, token, userAgent)) :
-        Retrofit2SyncCall.executeCall(registryService.getTags(repository, token, userAgent))
+      path ? Retrofit2SyncCall.executeCall(registryService.get(path, token, userAgent, queryParams)) :
+        Retrofit2SyncCall.executeCall(registryService.getTags(repository, token, userAgent, queryParams))
     }, repository)
 
     def nextPath = findNextLink(response?.headers())
     def tags = convertResponseBody(response.body(), DockerRegistryTags)
 
     if (nextPath) {
-      def nextTags = getTags(repository, nextPath)
+      def nextPathNew
+      (nextPathNew, queryParams) = parseForQueryParams(nextPath)
+      def nextTags = getTags(repository, nextPathNew, queryParams)
       tags.tags.addAll(nextTags.tags)
     }
 
     return tags
   }
 
+  static def parseForQueryParams(String nextPath) {
```

Comment:
javadoc please.  And please use a more specific return type even if groovy doesn't require it.

Reply:
Done!